application/controllers/collection_instrument.py

```diff
@@ -222,17 +231,22 @@ def upload_instrument_with_no_collection_exercise(self, survey_id, classifiers=N
         log.info("Upload instrument", survey_id=survey_id)
 
         validate_uuid(survey_id)
-        instrument = InstrumentModel(ci_type="EQ")
-
         survey = self._find_or_create_survey_from_survey_id(survey_id, session)
+        survey_service_details = get_survey_service_details(survey.survey_id)
+        ci_type = "EQ"
+        instrument = InstrumentModel(ci_type=ci_type)
         instrument.survey = survey
-
         if classifiers:
-            deserialized_classifiers = loads(classifiers)
+            classifiers = loads(classifiers)
+            if survey_service_details["surveyMode"] == "EQ_AND_SEFT":
```

Comment:
This is debatable, checking the survey_mode is not necessarily needed as you could do it on everything. I included it in the SEFT one because survey_service_details was in scope and therefore an extra call was not needed. Survey service is not in scope here and I have had to get it, which begs the question should we be doing it